packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionList.ts

```diff
@@ -0,0 +1,86 @@
+import { SubscriptionListItem } from '@/components/WfoSubscriptionsList';
+import { CacheTags, orchestratorApi } from '@/rtk';
+import {
+    BaseGraphQlResult,
+    GraphqlQueryVariables,
+    Subscription,
+    SubscriptionsResult,
+} from '@/types';
+
+export const subscriptionListQuery = `query SubscriptionsList(
+        $first: Int!
+        $after: Int!
+        $sortBy: [GraphqlSort!]
+        $filterBy: [GraphqlFilter!]
+        $query: String
+    ) {
+        subscriptions(
+            first: $first
+            after: $after
+            sortBy: $sortBy
+            filterBy: $filterBy
+            query: $query
+        ) {
+            page {
+                note
+                startDate
+                endDate
+                description
+                insync
+                status
+                subscriptionId
+                product {
+                    name
+                    tag
+                    productType
+                }
+                customer {
+                    fullname
+                    shortcode
+                }
+            }
+            pageInfo {
+                totalItems
+                startCursor
+                hasPreviousPage
+                hasNextPage
+                endCursor
+                sortFields
+                filterFields
+            }
+        }
+    }
+`;
+
+export type SubscriptionListResponse = {
+    subscriptions: Subscription[];
+} & BaseGraphQlResult;
+
+const subscriptionListApi = orchestratorApi.injectEndpoints({
+    endpoints: (builder) => ({
+        getSubscriptionList: builder.query<
+            SubscriptionListResponse,
+            GraphqlQueryVariables<SubscriptionListItem>
+        >({
+            query: (variables) => ({
+                document: subscriptionListQuery,
+                variables,
+            }),
+            transformResponse: (
+                response: SubscriptionsResult,
+            ): SubscriptionListResponse => {
+                const subscriptions = response.subscriptions.page || [];
```

Comment:
More of a general comment but I would map all the logic that converts the graphql result into a SubscriptionList Item (like applying mapGraphQlSubscriptionsResultToSubscriptionListItems) to this function. So you get items you can use directly from the hook

Reply:
I would not move such logic in here. Data from the backend should stay roughly in its shape. SubscriptionListItems are tailored for use in the table, typical view data. Adding that mapper here fully transforms the result from the graphql and makes possible reuse of this endpoint hard.